scripts/core/EXT_Exp_SubAllocationProperties.rst

```diff
@@ -0,0 +1,70 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+:orphan:
+
+.. _ZE_experimental_sub_allocations:
+
+=====================================
+ Sub-Allocation Properties Extension
+=====================================
+
+API
+----
+
+* Enumerations
+
+
+    * ${x}_sub_allocations_exp_version_t
+
+
+* Structures
+
+
+    * ${x}_memory_sub_allocations_exp_properties_t
+
+Sub-Allocation Properties
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Driver implementations may choose to create a device allocation as a series of sub-allocations. For instance,
+an allocation created against a parent device may be allocated internally as set of N sub-allocations, with
+N being the number of sub-devices associated with the parent device.
+
+The sub-allocation properties extension may be used to get the properties, i.e. base address and size, for
+each of those sub-allocations. The following pseudo-code demonstrates a basic use-case of this extension:
+
+.. parsed-literal::
+
+    ${x}MemAllocDevice(context, &desc, size, alignment, device, &ptr);
+
+    ${x}_memory_sub_allocations_exp_properties_t subAllocationDesc {};
+    uitn32_t numberOfSuballocations = 0;
+    subAllocationDesc.stype = ${X}_STRUCTURE_TYPE_MEMORY_SUB_ALLOCATIONS_EXP_PROPERTIES;
+    subAllocationDesc.pCount = &numberOfSuballocations;
```

Comment:
One final comment - this doesn't need to change now, but it's something to consider, especially if this ever becomes a non-experimental extension: We debatably have an unnecessary level of indirection here.  Because this entire structure is already an in-out structure we could simply return the "count" here, rather than returning via a "pointer-to-count".

Reply:
thanks @bashbaug . Yes, you are right there. We are following here the two-pass query model we have in other interfaces, but that we could remove one level of indirection at least. Yes, let's keep in mind depending on how this evolves based on feedback.